### PR TITLE
Drop Python 3.8 support (remove typing_extensions from depends)

### PR DESCRIPTION
**Pull request by yarikoptic**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
           - ubuntu-latest
           - macos-12
         python:
-          - 3.8
+          - 3.9
```
Comment (candleindark) [inline .github/workflows/test.yml:23]:
```suggestion
          - 3.9
```
We can just remove the `- 3.8` here since the 3.9 options is already available.

Comment (yarikoptic) [reply .github/workflows/test.yml:23]: ooops -- missed that!  we just need to remove this line since followed by 3.9... will do, thanks

Comment (yarikoptic) [reply .github/workflows/test.yml:23]: rright, removal suggestion would be 
```suggestion
```

**Comment by yarikoptic** [comment]
whatever had to run seems to have ran and is green!  there is some odd memory of 3.8 and macos-12 which are no more:

```
❯ git grep -e macos-12 -e '3\.8' | grep -v CHANG
❯ 
```
Let's proceed so other PRs could get greener too

**Pull request merged**